src/towncrier/test/helpers.py

```diff
@@ -69,3 +70,43 @@ def setup_simple_project(
 
     if mkdir_newsfragments:
         Path("foo/newsfragments").mkdir()
+
+
+def with_isolated_simple_project(
```

Comment:
Thanks for the refactoring.

I know that we already have `setup_simple_project`

The test decorators are all about isolation.

And not sure if "simple" has value.

Do we also have "with_isolated_complex_project" ?

```suggestion
def with_project(
```

Can we have a docstring for this function?

I think that in the docstring we can have more details about what is considered "simple".

My main goal for the docstring is to describe the intended use cases for each argument.

With that, you can decide if you need to extend this function, or create a new one (that for example is more complex)